feat(docker): add Docker support

seanmonstar opened the pull request:
WIP...

jbuck commented on docker-compose.yml:
```diff
@@ -0,0 +1,10 @@
+version: '2'
```
I'd agree with your/Ryans assertion that we can delete for this now.

vladikoff commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM ubuntu:trusty
```
👍

seanmonstar commented:
@jbuck ok, updated. Added a circle.yml, and added integration. I'm noticing that the build is failing cause it's not published to the docker library. Is that required to run the tests? I'd think that running this on circle would use the code in the repo, not something already published...

jbuck commented on Dockerfile-test:
```diff
@@ -0,0 +1,2 @@
+FROM fxa-custom-server:build
```
The reason it's trying to pull from docker hub is the missing `s` here, should be `fxa-customs-server`

seanmonstar commented:
![its working!!](http://i.imgur.com/OWT9Tud.gif)

jbuck commented on Dockerfile-build:
```diff
@@ -0,0 +1,14 @@
+FROM mhart/alpine-node:4.7.2
```
Lets update this to 4.8.0 since that's out now

seanmonstar commented:
Updated the alpine-node version to 4.8.0.
